### PR TITLE
[rps] Replace TSLint with ESlint in RPS

### PR DESCRIPTION
**Pull request by Alonski**
### Description

This PR replaces TSLint with ESlint in RPS.
TSLint is deprecated.

This also setups Prettier to lint SCSS and JSON files which can't be checked by ESLint, from what I could figure out.

The rules can be changed.
What does everyone think?

### CODE REVIEW

**Comment by Alonski** [comment]
I managed to get VSCode + ESLint + Prettier playing nicely together :)
This is my `.vscode/settings.json` files:
```json
{
  "eslint.validate": ["javascript", "javascriptreact", "typescript", "typescriptreact"],
  "eslint.workingDirectories": [{"directory": "packages/rps", "changeProcessCWD": true}],
  "editor.formatOnSave": true
}
```
We will need to setup the `eslint.workingDirectories` option for each package.

I also downgraded Prettier in RPS to v1.18 so its consistent with other packages.
I also changed `ESLintrc` to not define Prettier rules but instead to inherit them from `.pretterrc`

**Comment by tomclose** [comment]
Could you add your `vscode/settings.json` to the repo? I think we should definitely be sharing settings like these, if the development experience is broken without them.

File: packages/rps/package.json [hunk 1]
```diff
@@ -15,11 +15,13 @@
...
-    "lint:check": "tslint -c tslint.json -p .",
-    "lint:write": "tslint -c tslint.json -p . --fix",
+    "lint:check": "run-s eslint:check prettier:check",
+    "lint:write": "run-s eslint:write prettier:write",
+    "eslint:check": "eslint \"**/*.{js,ts,tsx}\" --cache",
+    "eslint:write": "eslint \"**/*.{js,ts,tsx}\" --fix",
     "postinstall": "rm -f node_modules/web3/index.d.ts",
-    "prettier:check": "prettier --check '**/*.{scss,json,js,ts,tsx}' --ignore-path '.gitignore'",
-    "prettier:write": "prettier --write '**/*.{scss,json,js,ts,tsx}' --ignore-path '.gitignore'",
+    "prettier:check": "prettier --check '**/*.{scss,json}' --ignore-path '.gitignore'",
```
Comment (snario) [inline packages/rps/package.json]: I think `eslint` is running `prettier` so we shouldn't need special `prettier:check` scripts

File: packages/rps/package.json [hunk 2]
```diff
@@ -34,7 +36,7 @@
     "precommit": "lint-staged --quiet"
   },
   "lint-staged": {
-    "{src,tests}/**/*.{ts,tsx}": "yarn lint:check"
+    "{src,tests}/**/*.{ts,tsx}": "eslint"
```
Comment (snario) [inline packages/rps/package.json]: Since you're using `.eslintignore` and `**/*.{js,ts,tsx}` for the linting, maybe you should use that regex here too.

File: packages/rps/package.json (hunk 1 above)
Comment (Alonski) [reply packages/rps/package.json]: ESLint, from what I could manage to figure out, can't run on `scss` and `json` files
I want to run Prettier on those files though :)

File: packages/rps/package.json (hunk 2 above)
Comment (snario) [inline packages/rps/package.json]: Also I recommend making `prettier` a `warn` and then doing `eslint --max-warnings=0`. I find it nice to separate linting issues versus build issues with yellow and red squiggly lines respectively in my editor.

Comment (Alonski) [reply packages/rps/package.json]: So I should change this back to `yarn lint:check`?

Comment (Alonski) [reply packages/rps/package.json]: So it checks Prettier as well?

Comment (Alonski) [reply packages/rps/package.json]: Oh nevermind you meant on the left side :)

File: packages/rps/src/utils/channel-client.ts [hunk 3]
```diff
@@ -61,7 +61,7 @@ type UnsubscribeFunction = () => void;
 // The message Payload is designed to be opaque to the app. However, it's useful
 // to be able to specify the Payload type for the FakeChannelClient, as we'll be
 // manipulating it within the client.
-export interface IChannelClient<Payload = any> {
+export interface ChannelClient<Payload = any> {
```
Comment (geoknee) [inline packages/rps/src/utils/channel-client.ts]: Why was this necessary?  I would have thought the prefixed `I` was pretty helpful in distinguishing this interface from some class that implements it.

**Comment by geoknee** [comment]
Should we remove `tslint` and associated configuration as a part of this PR?

**Comment by geoknee** [comment]
See #696

File: packages/rps/src/utils/channel-client.ts (hunk 3 above)
Comment (Alonski) [reply packages/rps/src/utils/channel-client.ts]: I see you resolved this :)

**Comment by Alonski** [comment]
Thanks @geoknee!
